services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/model/MongoDbBasedTenantDao.java

```diff
@@ -378,6 +378,35 @@ public Future<String> update(
                 .onComplete(r -> span.finish());
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Future<Integer> count(final JsonObject filter, final SpanContext tracingContext) {
+
+        final Span span;
+        if (tracingContext != null) {
+            span = tracer.buildSpan("get Tenants count")
+                    .addReference(References.CHILD_OF, tracingContext)
+                    .start();
+        } else {
+            span = null;
```

Comment:
how about using a *NoopSpan* here instead of `null`? We can then simply always finish the span at the end of the method ...